chain/chain/src/store.rs

```diff
@@ -1745,12 +1741,13 @@ impl<'a> ChainStoreUpdate<'a> {
         block_hash: CryptoHash,
         block_merkle_tree: PartialMerkleTree,
     ) {
-        self.chain_store_cache_update
-            .block_ordinal_to_hash
-            .insert(block_merkle_tree.size(), block_hash);
         self.chain_store_cache_update.block_merkle_tree.insert(block_hash, block_merkle_tree);
     }
 
+    pub fn save_block_ordinal(&mut self, block_hash: CryptoHash, block_ordinal: NumBlocks) {
+        self.chain_store_cache_update.block_ordinal_to_hash.insert(block_ordinal, block_hash);
```

Comment:
I'm not very familiar with the code. Is this enough so that this change is ultimately reflected on the database?

Reply:
Yes. It is saved when we call `ChainStoreUpdate::commit`